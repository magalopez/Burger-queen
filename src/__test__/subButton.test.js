import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import "@testing-library/jest-dom/extend-expect";
import SubButton from "../components/generic_components/subButtons";

afterEach(cleanup);

const products = [
  [
    { id: 1, title: "uno", price: 11, counter: 1 },
    { id: 2, title: "dos", price: 22, counter: 2 }
  ]
];

const addProd = jest.fn((id, title, price, counter) => {
  const newProducts = [{ id, title, price, counter }];
  return newProducts;
});

test("renders", () => {
  const { asFragment } = render(
    <SubButton productElement={products} addProduct={addProd} />
  );

  const firtsRender = asFragment();
  expect(firtsRender).toMatchSnapshot();
});

test("", () => {
  const { getByTestId } = render(
    <SubButton productElement={products} addProduct={addProd} />
  );

  const ancestor = getByTestId("subbuttontag");
  const descendant = getByTestId("buttontag");

  expect(ancestor).toContainElement(descendant);
});
