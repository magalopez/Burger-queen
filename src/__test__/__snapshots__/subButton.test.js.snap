// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="tile is-child buttons is-12 is-centered columns is-mobile"
    data-testid="subbuttontag"
  >
    <div
      class="field is-grouped"
    >
      <button
        class="button is-outlined "
        data-testid="buttontag"
      />
    </div>
  </div>
</DocumentFragment>
`;
